otx/algorithms/classification/adapters/mmcls/models/heads/custom_cls_head.py

```diff
@@ -15,16 +15,12 @@
 class CustomNonLinearClsHead(NonLinearClsHead):
     """Custom Nonlinear classifier head."""
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.loss_type = kwargs.get("loss", dict(type="CrossEntropyLoss"))["type"]
-
     def loss(self, cls_score, gt_label, feature=None):
         """Calculate loss for given cls_score/gt_label."""
         num_samples = len(cls_score)
         losses = dict()
         # compute loss
-        if self.loss_type == "IBLoss":
+        if self.compute_loss._get_name() == "IBLoss":
```

Comment:
Is the function called _get_name a function that can be called from any compute_loss?

Reply:
I think you should avoid protected function also.

Reply:
As far as I know, yes. There is a TC in E2E test. https://github.com/openvinotoolkit/training_extensions/blob/develop/tests/e2e/cli/classification/test_classification.py#L111-L117 `TestToolsMultiClassClassification::test_otx_train` does incremental learning, IBLoss returns well "IBLoss". Ah, CrossEntropyLoss class also returns well "CrossEntropyLoss"

Reply:
Updated in 02e57c70fad225ad826c6ee492f8e46ee87a8e1f to use `self.compute_loss.__class__.__name__`